google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITBucketTest.java

```diff
@@ -0,0 +1,379 @@
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ITBucketTest {
+
+  @ClassRule(order = 1)
+  public static final StorageFixture storageFixture = StorageFixture.defaultHttp();
+
+  @ClassRule(order = 2)
+  public static final BucketFixture bucketFixture =
+      BucketFixture.newBuilder().setHandle(storageFixture::getInstance).build();
+
+  @ClassRule(order = 3)
+  public static final BucketFixture requesterPaysFixture =
+      BucketFixture.newBuilder().setHandle(storageFixture::getInstance).build();
+
+  private static Storage storage;
+  private static String bucketName;
+  private static String requesterPaysBucketName;
+
+  private static final byte[] BLOB_BYTE_CONTENT = {0xD, 0xE, 0xA, 0xD};
+  private static final Map<String, String> BUCKET_LABELS = ImmutableMap.of("label1", "value1");
+  private static final Long RETENTION_PERIOD = 5L;
+
+  @BeforeClass
+  public static void setup() {
+    storage = storageFixture.getInstance();
+    bucketName = bucketFixture.getBucketInfo().getName();
+    requesterPaysBucketName = requesterPaysFixture.getBucketInfo().getName();
+  }
+
+  @AfterClass
+  public static void cleanup() {
+    unsetRequesterPays();
+  }
+
+  @Test(timeout = 5000)
+  public void testListBuckets() throws InterruptedException {
+    Iterator<Bucket> bucketIterator =
+        storage
+            .list(Storage.BucketListOption.prefix(bucketName), Storage.BucketListOption.fields())
+            .iterateAll()
+            .iterator();
+    while (!bucketIterator.hasNext()) {
+      Thread.sleep(500);
+      bucketIterator =
+          storage
+              .list(Storage.BucketListOption.prefix(bucketName), Storage.BucketListOption.fields())
+              .iterateAll()
+              .iterator();
+    }
+    while (bucketIterator.hasNext()) {
+      Bucket remoteBucket = bucketIterator.next();
+      assertTrue(remoteBucket.getName().startsWith(bucketName));
+      assertNull(remoteBucket.getCreateTime());
+      assertNull(remoteBucket.getSelfLink());
+    }
+  }
+
+  @Test
+  public void testGetBucketSelectedFields() {
+    Bucket remoteBucket = storage.get(bucketName, Storage.BucketGetOption.fields(BucketField.ID));
+    assertEquals(bucketName, remoteBucket.getName());
+    assertNull(remoteBucket.getCreateTime());
+    assertNotNull(remoteBucket.getGeneratedId());
+  }
+
+  @Test
+  public void testGetBucketAllSelectedFields() {
+    Bucket remoteBucket =
+        storage.get(bucketName, Storage.BucketGetOption.fields(BucketField.values()));
+    assertEquals(bucketName, remoteBucket.getName());
+    assertNotNull(remoteBucket.getCreateTime());
+    assertNotNull(remoteBucket.getSelfLink());
+  }
+
+  @Test
+  public void testBucketLocationType() {
+    storage.update(BucketInfo.newBuilder(bucketName).setLocation("us").build());
+    Bucket bucket = storage.get(bucketName);
+    assertEquals("multi-region", bucket.getLocationType());
```

Comment:
Not something that needs to change in this PR, however these kinds of tests which actually mutate the bucket state should be refactored to create their own bucket so they are hermetic.

Reply:
can do